lua/controller.lua

```diff
@@ -297,6 +297,79 @@ local function controller_insert_to_drawers(pos, stack)
 	return stack
 end
 
+--[[
+	Returns an array of drawers in the drawer network with their positions and slot
+	data. Slot data includes stored item, item count, and max count.
+
+	Results can be paginated by specifying an `offset` and a `max_count`.
+]]
+local function controller_get_network_info(pos, offset, max_count)
+	local found_drawers = {}
+	local index = index_drawers(pos)
+
+	-- Add each drawer in the network separately
+	-- Sort them by their positions to keep order for pagination
+	local drawer_positions = {}
+	local drawer_count = 0
+	local keys = {}
+	for _, drawer in pairs(index) do
+		local position = drawer.drawer_pos
+		local key = vector.to_string(position)
+
+		if not drawer_positions[key] then
+			drawer_positions[key] = position
+			table.insert(keys, key)
+			drawer_count = drawer_count + 1
+		end
+	end
+	table.sort(keys)
+
+	for _, key in ipairs(keys) do
+		local position = drawer_positions[key]
+		local node = core.get_node(position)
+		local drawer_meta = core.get_meta(position)
+		local node_def = core.registered_nodes[node.name]
+		local drawer_type = node_def.groups.drawer
+
+		-- Record information of each slot
+		local slots = {}
+		for i = 1, drawer_type do
+			-- 1x1 drawers don't have numbers in the meta fields
+			local slot_id = (drawer_type == 1 and "") or i
+			local slot_name = drawer_meta:get_string("name" .. slot_id)
+			local slot_count = drawer_meta:get_int("count" .. slot_id)
+			local slot_max = drawer_meta:get_int("max_count" .. slot_id)
+
+			table.insert(slots, {
+				name = slot_name,
+				count = slot_count,
+				max = slot_max
+			})
+		end
+
+		table.insert(found_drawers, {position=position, slots=slots})
+	end
+
+	-- Offset must be an integer and >= 1
+	offset = (type(offset) == "number") and
+		math.max(1, math.floor(offset)) or 1
```

Comment:
This seems a bit complicated. Why not...
```suggestion
	offset = math.max(1, math.floor(tonumber(offset) or 1))
```
likewise for `max_count`:
```lua
	max_count = tonumber(max_count) or drawers.CONTROLLER_MAX_MATCHES
	max_count = math.min(math.max(1, math.floor(max_count)), drawers.CONTROLLER_MAX_MATCHES)
```

Reply:
I borrowed some code from the `digiline_craftdb` mod for pagination.